Stratis.Bitcoin/Logging/LogsExtension.cs

```diff
@@ -7,99 +7,181 @@
 using Stratis.Bitcoin.Configuration;
 using Stratis.Bitcoin.Configuration.Settings;
 using Stratis.Bitcoin.Utilities;
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+using System.Text;
 
 namespace Stratis.Bitcoin.Logging
 {
+    /// <summary>
+    /// Integration of NLog with Microsoft.Extensions.Logging interfaces.
+    /// </summary>
     public static class LogsExtension
     {
-        public static void AddFilters(this ILoggerFactory loggerFactory, LogSettings settings)
+        /// <summary>Width of a column for pretty console/log outputs.</summary>
+        public const int ColumnLength = 16;
+
+        /// <summary>Currently used node's log settings.</summary>
+        private static LogSettings logSettings;
+
+        /// <summary>Currently used data folder to determine path to logs.</summary>
+        private static DataFolder folder;
+
+        /// <summary>
+        /// Initializes application logging.
+        /// </summary>
+        static LogsExtension()
+        {
+            // If there is no NLog.config file, we need to initialize the configuration ourselves.
+            if (LogManager.Configuration == null) LogManager.Configuration = new LoggingConfiguration();
+
+            // Installs handler to be called when NLog's configuration file is changed on disk.
+            LogManager.ConfigurationReloaded += NLogConfigurationReloaded;
```

Comment:
Yes, most of the things you don't like here are solely for the adhoc reloading. This is for developers only. It does not happen very often, but you are running node and it is running very long time and has some very interesting context and then you find out that you are missing  some logs (or you think you are seeing too many logs, or you want some logs to put into separate file, etc.), so what you can do is to open the config file (it is actually described in the using-logging.md) and change it without stopping your node. This will cause this handler to be called. 

Now this handler is currently static method. As such, it needs those static fields. 

So an option is not to have this handler as a static method, but then we would need somehow to get access to NodeSettings. So What we could do is to put this handler to NodeSettings, which would make the dataFolder and LogsSettings accessible. And we would have to install the handler when we have NodeSettings and not before.

Do you think it is worth and better than having those static things?